e2e-tests/src/main/java/gov/nasa/jpl/aerie/e2e/procedural/scheduling/procedures/ExternalEventsSourceQueryGoal.java

```diff
@@ -0,0 +1,39 @@
+package gov.nasa.jpl.aerie.e2e.procedural.scheduling.procedures;
+
+import gov.nasa.ammos.aerie.procedural.scheduling.Goal;
+import gov.nasa.ammos.aerie.procedural.scheduling.annotations.SchedulingProcedure;
+import gov.nasa.ammos.aerie.procedural.scheduling.plan.EditablePlan;
+import gov.nasa.ammos.aerie.procedural.timeline.payloads.ExternalSource;
+import gov.nasa.ammos.aerie.procedural.timeline.payloads.activities.DirectiveStart;
+import gov.nasa.ammos.aerie.procedural.timeline.plan.EventQuery;
+import gov.nasa.jpl.aerie.merlin.protocol.types.SerializedValue;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Map;
+
+@SchedulingProcedure
+public record ExternalEventsSourceQueryGoal() implements Goal {
+  @Override
+  public void run(@NotNull final EditablePlan plan) {
+
+    // extract events belonging to the second source
+    EventQuery eventQuery = new EventQuery(
+        null,
+        null,
+        List.of(new ExternalSource("NewTest.json", "TestGroup_2"))
```

Comment:
Is it safe to assume that users will know external event source file names _a priori_? Or would it be helpful to include some way to query for which sources had been registered with a plan?

I'm thinking of cases where source file names might be autogenerated or have some unknown timestamp, and users might just want to reference events where `source.includes("DSN")` or something.

While typing this out, I'm realizing that particular use case could be accomplished by "manually" filtering on event sources after we've run the `EventQuery` against the `Plan` 🤔

Reply:
Not necessarily, but it is possible to allow users to filter for that kind of thing: 
```
var events = plan.events();
events.filter(false, event -> event.source.key.contains("DSN"));
```

Reply:
My thoughts exactly!